components/Comment/CommentEditor.tsx

```diff
@@ -1,62 +1,62 @@
+import { useEffect, useRef, useState } from "react";
+import Button from "../Form/Button";
+import CreateBountyBtn from "../Bounty/CreateBountyBtn";
+import ReactDOMServer from "react-dom/server";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faVideo } from "@fortawesome/free-solid-svg-icons";
+import buildQuillModules from "./lib/buildQuillModules";
+import QuillFormats from "./lib/quillFormats";
+import isQuillEmpty from "../TextEditor/util/isQuillEmpty";
+import { AuthorProfile } from "~/config/types/root_types";
+import CommentAuthors from "./CommentAuthors";
 
-const ReactQuill = dynamic(() => import("react-quill"), { ssr: false });
 
 type CommentEditorArgs = {
-  previewWhenInactive?: boolean;
+  editorId: string,
   placeholder?: string;
   handleSubmit: Function;
-  content?: string;
+  content?: object;
   allowBounty?: boolean;
+  author?: AuthorProfile | null;
 };
 
 const CommentEditor = ({
-  previewWhenInactive = false,
+  editorId,
   placeholder = "Add comment or start a bounty",
   handleSubmit,
-  content = "",
+  content = {},
   allowBounty = false,
+  author,
 }: CommentEditorArgs) => {
   const editorRef = useRef<any>(null);
-  const [isSubmitDisabled, setIsSubmitDisabled] = useState(true);
-  const [_previewWhenInactive, _setPreviewWhenInactive] =
-    useState(previewWhenInactive);
-  const [_isPreview, _setIsPreview] = useState(previewWhenInactive);
-  const _isPreviewRef = useRef(_isPreview);
-  const [_content, _setContent] = useState<string>(content);
-  const [_isFocused, _setIsFocused] = useState(false);
-
-  const handleEditorChange = (value, delta, source, editor) => {
-    const editorContents = editor.getContents();
-    if (isQuillEmpty(editorContents)) {
-      setIsSubmitDisabled(true);
-    } else {
-      setIsSubmitDisabled(false);
-    }
-
-    _setContent(value);
-  };
+  const [isSubmitDisabled, setIsSubmitDisabled] = useState<boolean>(true);
+  const [_content, _setContent] = useState<object>(content);
+  const contentRef = useRef<object>(content);
+  const [isFullToolbarOpen, setIsFullToolbarOpen] = useState<boolean>(false);
+  const [isFocused, setIsFocused] = useState<boolean>(false);
+  const { quill, quillRef, Quill } = useQuill({
+    modules: buildQuillModules({
+      editorId,
+      handleImageUpload: () => null,
+      handleSubmit: () => handleSubmit({ content: contentRef })
+    }),
+    formats: QuillFormats
+  });
 
   useEffect(() => {
     const _handleClick = (e) => {
-      const isOutsideClick =
-        !_isPreviewRef.current && !editorRef.current?.contains(e.target);
-      const excludedElems = [".reply-btn", ".edit-btn"];
-      const clickOnExcluded = excludedElems.reduce(
-        (prev, selector) => Boolean(prev || e.target.closest(selector)),
-        false
-      );
+      const isFullToolbarTriggerClick = e.target.closest(".show-full-editor");
+      const isFullToolbarClick = e.target.closest(".ql-full-editor");
```

Comment:
Is it possible to get this inside the react components instead of building handler here? 

For example in a <div> below, if we can do 

```
<div onClick onEnter onLeave

```

Or something like that instead. Not sure it's possible, just wondering

Reply:
I ended up doing something like this by moving the state of the toolbar to within the toolbar. Code is cleaner now on both `CommentEditor` and `CommentEditorToolbar`